src/lib/span.ts

```diff
@@ -0,0 +1,38 @@
+export class Span {
+    spanId: string;
+    name: string;
+    startTime: number;
+    endTime: number | null;
+    parentSpanId: string | null;
+    children: Span[];
+
+    constructor(name: string, parentSpanId: string | null = null) {
+        this.spanId = `span-${Date.now()}-${Math.random().toString(36).substr(2, 5)}`;
+        this.name = name;
+        this.startTime = Date.now();
+        this.endTime = null;
+        this.parentSpanId = parentSpanId;
+        this.children = [];
+    }
+
+    close(): void {
+        this.endTime = Date.now();
+    }
+
+    isCompleted(): boolean {
+        return this.endTime !== null;
+    }
```

Comment:
All function definitions should have the `public` or `protected` keyword explicitly provided.